### PR TITLE
Removed LegacyVersion from ontopy.utils

### PR DESCRIPTION
**Pull request by francescalb**
# Description:

## Type of change:

- [x] Bug fix.
- [ ] New feature.
- [ ] Documentation update.

## Checklist:

This checklist can be used as a help for the reviewer.

- [ ] Is the code easy to read and understand?
- [ ] Are comments for humans to read, not computers to disregard?
- [ ] Does a new feature has an accompanying new test (in the CI or unit testing schemes)?
- [ ] Has the documentation been updated as necessary?
- [ ] Does this close the issue?
- [ ] Is the change limited to the issue?
- [ ] Are errors handled for all outcomes?
- [ ] Does the new feature provide new restrictions on dependencies, and if so is this documented?

### CODE REVIEW

**Comment by francescalb** [comment]
LegacyVersion was removed from the package 'packaging'. It was noly used for type_checking, so removing should not be a problem.

**Review by CasperWA — COMMENTED** [review]
I don't think there is a need to remove `LegacyVersion` here?
Or does it not exist any more in the newest version of `packaging`?

**Comment by CasperWA** [comment]
> I don't think there is a need to remove `LegacyVersion` here? Or does it not exist any more in the newest version of `packaging`?

Ah. I should just read your comment. Didn't find a reason for the PR at first because it wasn't in the PR description but buried underneath, sorry.

File: ontopy/utils.py [hunk 1]
```diff
@@ -447,16 +447,16 @@ def _validate_installed_version(
...
     if isinstance(min_version, str):
         min_version = parse_version(min_version)
-    elif isinstance(min_version, (LegacyVersion, Version)):
+    elif isinstance(min_version, (Version)):
         # We have the format we want
         pass
     else:
         raise TypeError(
-            "min_version should be either a str, LegacyVersion or Version. "
+            "min_version should be either a str (Version). "
```
Comment (CasperWA) [inline ontopy/utils.py:459]:
```suggestion
            "min_version should be either a str or packaging.Version. "
```

File: ontopy/utils.py [hunk 2]
```diff
@@ -447,16 +447,16 @@ def _validate_installed_version(
     """
     # pylint: disable=import-outside-toplevel
     import importlib
-    from packaging.version import parse as parse_version, LegacyVersion, Version
+    from packaging.version import parse as parse_version, Version
 
     if isinstance(min_version, str):
         min_version = parse_version(min_version)
-    elif isinstance(min_version, (LegacyVersion, Version)):
+    elif isinstance(min_version, (Version)):
```
Comment (CasperWA) [inline ontopy/utils.py:454]:
```suggestion
    elif isinstance(min_version, Version):
```
This is not going to work if `Version` is only imported in the `TYPE_CHECKING` block!

File: ontopy/utils.py [hunk 3]
```diff
@@ -21,7 +21,7 @@
 
 
 if TYPE_CHECKING:
-    from packaging.version import Version, LegacyVersion
+    from packaging.version import Version
```
Comment (CasperWA) [inline ontopy/utils.py:24]: Move this up and out of the `TYPE_CHECKING` block as `Version` is used explicitly in line 454 and not just as a type hint.

File: ontopy/utils.py [hunk 4]
```diff
@@ -447,16 +447,16 @@ def _validate_installed_version(
     """
     # pylint: disable=import-outside-toplevel
     import importlib
-    from packaging.version import parse as parse_version, LegacyVersion, Version
+    from packaging.version import parse as parse_version, Version
```
Comment (CasperWA) [inline ontopy/utils.py:450]:
```suggestion
    from packaging.version import parse as parse_version
```

Comment (CasperWA) [reply ontopy/utils.py:450]: Import `Version` and the top of the file instead.

**Review by CasperWA — CHANGES_REQUESTED** [review]

**Review by CasperWA — APPROVED** [review]

**Pull request merged**